### PR TITLE
feat: accept metric names without `neg_` prefix in `summarize()`

### LINKED ISSUE #2607: feat: Make it possible to pass sklearn metrics without `"neg_"`
### Is your feature request related to a problem? Please describe.

The following works:

```python
from sklearn.datasets import make_classification
from sklearn.ensemble import RandomForestClassifier
from skore import evaluate

X, y = make_classification(random_state=0)
report = evaluate(
    RandomForestClassifier(max_depth=2, random_state=0), X, y, pos_label=1
)
report.metrics.summarize(metric='neg_mean_squared_error').frame()
```

But this doesn't:
```python
report.metrics.summarize(metric='mean_squared_error').frame()
```

### Describe the solution you'd like

Make it work

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by direkkakkar319-ops**
## Description

Fixes #2607
When using `report.metrics.summarize()`, passing error metric names without the `neg_` prefix (e.g. `"mean_squared_error"`) would raise a `ValueError` because sklearn's scorer registry only knows the `neg_` .

## Root Cause

Sklearn follows a "higher is better" convention for all scorers.
Error metrics like `mean_squared_error` are therefore only registered as `neg_mean_squared_error` in sklearn's scorer registry.

## Solution

Added a `_METRIC_ALIASES` dictionary inside `_MetricsAccessor` that maps user-friendly metric names to their `neg_` prefixed sklearn scorer equivalents.
```python
_METRIC_ALIASES: dict[str, str] = {
    "mean_squared_error": "neg_mean_squared_error",
    ...
}
```

## Changes

- `skore/src/skore/_sklearn/_estimator/metrics_accessor.py` — added `_METRIC_ALIASES` dict and resolution logic in `summarize()`
- In `skore/src/skore/_sklearn/_estimator/metrics_accessor.py`
made this code 

image

to 

image

- `skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py` — added test cases for the fix
- `sphinx/changelog.rst` — updated changelog

## Testing
```python
# Before this fix — raised ValueError
report.metrics.summarize(metric='mean_squared_error').frame()

# After this fix-working correct
report.metrics.summarize(metric='mean_squared_error').frame()
report.metrics.summarize(metric='neg_mean_squared_error').frame()  # still works too
```

Screenshot 2026-03-21 203902

### CODE REVIEW

**Comment by direkkakkar319-ops** [comment]
was duplicate of #2646 so it was closed
